core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -601,8 +602,11 @@ private Query parse(AliasMetaData alias, QueryShardContext shardContext) {
         try {
             byte[] filterSource = alias.filter().uncompressed();
             try (XContentParser parser = XContentFactory.xContent(filterSource).createParser(filterSource)) {
-                ParsedQuery parsedFilter = shardContext.toFilter(shardContext.newParseContext(parser).parseInnerQueryBuilder());
-                return parsedFilter == null ? null : parsedFilter.query();
+                Optional<QueryBuilder> innerQueryBuilder = shardContext.newParseContext(parser).parseInnerQueryBuilder();
+                if (innerQueryBuilder.isPresent()) {
+                    return shardContext.toFilter(innerQueryBuilder.get()).query();
+                }
+                return null;
```

Comment:
here as well, would be nice like above not to have to return null ever. I think if we make validation stricter then we can rely on the filter to always be present?